Dockerfile

```
@@ -37,8 +36,8 @@ COPY . .
 # Rayon threads are limited to minimize memory requirements in CI, avoiding OOM
 # Rust threads are increased with a nightly feature for faster compilation (single-threaded by default)
 ARG CUDA_COMPUTE_CAP=70
-ARG RAYON_NUM_THREADS=4
-ARG RUST_NUM_THREADS=4
+ARG RAYON_NUM_THREADS=64
+ARG RUST_NUM_THREADS=64
```

I think this was not intended as part of the PR? CI will not like this according to the comment above these lines:

```suggestion
ARG RAYON_NUM_THREADS=4
ARG RUST_NUM_THREADS=4
```